bundles/org.openhab.binding.semsportal/README.md

```diff
@@ -0,0 +1,45 @@
+# SEMSPortal Binding
+
+This binding can help you include statistics of your SEMS / GoodWe solar panel insallation into openHAB. It is a readonly connection that maps collected parameters to openhab channels. It provides current, day, month and total yields, as well as some income statistics if you have configured these in the SEMS portal. It requires a power station that is connected through the internet to the SEMS portal.
+
+## Supported Things
+
+There is only one thing type supported in this binding, which is the powerStation. You need to provide the identifying UUID of the solar system in your account yourself. If you have multiple powerstations that report to the same account, you can configure the same account for each powerSation and only differ the unique id of the station in each Thing.
+
+## Discovery
+
+At this moment there is no discovery possible. Configuration of the powerStation unique id is done by hand in the Thing configuration page.
+
+## Thing Configuration
+
+The configuration of the thing is pretty straight forward. You need to have your power station set up in the SEMS portal, and you need to have an account that is allowed to view the power station data. You should log in at least once in the portal with this account to activate it. 
+
+The thing needs the username and password to connect and retreive the data. It also needs the unique id of the power station. It can be found after you log in to the portal, on the URL. use the value you find in the URL at the x-es (including the -s) in this example: https://www.semsportal.com/powerstation/powerstatussnmin/xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+
+
+## Channels
+
+
+| channel       | type      | description                                                                                                |
+| ------------- | --------- | ---------------------------------------------------------------------------------------------------------- |
+| onlineState   | Switch    | If the powerStation is currently on and reporting to the portal                                            |
```

Comment:
I have a feeling this should change the ThingStatus and not be a separate channel.

Reply:
I think there's a difference: the openHAB Thing is still online as it has a connection with the SEMS portal. The communication with the SEMS portal is fine and we receive reports from it, but part of the report might be that the power station itself is currently shut down (because it turns off automatically when the power generation is low). In my interpretation, the Thing status indicates if openHAB is able to communicate with it, so it should remain online as long as the portal is returning answers and offline when it doesn't.

Reply:
Please understand the following is only my opinion and that I do not have the solor panels nor will I use the binding. I feel that if you can not get current up to date data for your station, then it should go OFFLINE. However I guess that would depend on how often it does that, if it happens every time a cloud goes over and it is >10 times a day, I would probably feel differently. I am just making a suggestion, you know it better than me and will use it :) If you feel it should not change then please press the 'Resolve conversation' button.

Reply:
Your remarks got me thinking and I think the aspect you really want to know about the power station is if it is operational or not. Therefore I renamed the Switch channel to "operational". I personally think a Thing with an Offline state is kind of an error you need to look into. That might be because of my limited experience with openHAB, I only use a few bindings where Offline is almost equivalent to "look into the hardware or the binding configuration as there is an error". That is why I'm hesitant to use the Offline state: the station is just asleep and you do not need to do anything.

Reply:
Indeed, this is exactly the purpose of the Thing status. It's fine if the Thing goes offline at night. You could add a note to the Thing status detail message, that it's in nightmode, if available.

Reply:
I'm sorry - a bit confused by this comment. Do you mean 1) the Thing Status should be "Offline" at night; or 2) the Thing Status should indicate "something wrong" and the solution I proposed - an "Operational" channel is ok?

Reply:
First. Just for clarification, the Thing status *can* be offline at night, but it don't *need* to be offline.

It's not an error if a Thing is offline. If it's not reachable for whatever reason (inteded or unintended), it should be offline.